docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -864,6 +864,24 @@ Cython code.  Here is the list of currently supported directives:
     division is performed with negative operands.  See `CEP 516
     <https://github.com/cython/cython/wiki/enhancements-division>`_.  Default is
     False.
+    
+``cpow`` (True / False)
+    ``cpow`` has two effects on the behaviour of ``a**b``.
+    
+    #. If both types are C integers then setting ``cpow`` to True 
+       forces the result type to be an integer. Otherwise the result 
+       type is a C floating point number if ``b`` can't be determined
+       to definitely be positive (the False case deviates from Python
+       slightly, since Python will dynamically pick either a Python
+       ``int`` or a Python ``float``).
+    #. If either or ``a`` or ``b`` is a floating point type then
+       setting ``cpow`` to True forces the result type to be a C
+       double, with a value of ``NaN`` if the result would be a
+       complex number. Otherwise the result can be either a real or
+       a complex number, at the cost of some speed.
```

Comment:
From reading this, it's not always clear where a distinction is made between the True/False cases and where you relate to something being positive, negative or complex for triggering a certain behaviour. Could you tear that a little more apart?

Reply:
I've replaced this with a table. I think that's the clearest way of presenting it in the generated documentation. (It may not be hugely easy to read it github though)